eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -237,6 +237,39 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       log.debug("restoring channel")
       context.system.eventStream.publish(ChannelRestored(self, data.channelId, peer, remoteNodeId, data))
       txPublisher ! SetChannelId(remoteNodeId, data.channelId)
+
+      // we watch all unconfirmed funding txs, whatever our state is
+      // (there can be multiple funding txs due to rbf, and they can be unconfirmed in any state due to zero-conf)
+      data.metaCommitments.commitments.foreach { commitment =>
+        commitment.localFundingStatus match {
+          case LocalFundingStatus.UnknownFundingTx =>
+            // Legacy single-funded channels. The funding tx may or may not be confirmed. If it was confirmed, the watch
+            // will trigger instantly, the state will be updated and a watch-spent will be set.
+
+            watchFundingConfirmed(commitment.fundingTxId, Some(singleFundingMinDepth(data)))
```

Comment:
I'm not very satisfied with the existence of `UnknownFundingTx`, but we can't handle it any other way due the `ConfirmedFundingTx` requiring a full transaction that we simply don't have for single-funding channels in most cases. Do we really need the full funding tx in the single-funding case? If not, maybe we could make it a `trait` and provide the full funding transaction only in the dual-funding case.

Reply:
I agree it's not very satisfying, but I think it's fine for now (especially for this PR).